solvation_analysis/solvation.py

```diff
@@ -114,7 +116,14 @@ def get_closest_n_mol(u, central_species, n_mol=5, radius=3):
     partial_shell = u.select_atoms(f"point {str_coords} {radius}")
     shell_resids = partial_shell.resids
     if len(np.unique(shell_resids)) < n_mol + 1:
-        return get_closest_n_mol(u, central_species, n_mol, radius + 1)
+        return get_closest_n_mol(
+            u,
+            central_species,
+            n_mol,
+            radius + 1,
+            return_resids=return_resids,
+            return_radii=return_radii,
+        )
     radii = distances.distance_array(coords, partial_shell.positions, box=u.dimensions)[
         0
```

Comment:
Yes this is coming from [Black](https://github.com/psf/black), which is PEP8 compliant. I am not super versed on the pros/cons of auto-formatting. I agree this looks strange but it's kinda nice to cede formatting control. Is there an official (or unofficial) MDAnalysis take on formatting style beyond using PEP8?

Reply:
Unofficial personal take: Black formats code for easy diffing rather than easy reading. I'm personally not a fan of it; IMO black removes many visual cues that make for easy skimming of the code. The reformatting highlighted here, for example, just has a random 0 on its own line without the [0] context that implies it's an index. Even had your line been a tad longer it would have been formatted like:
```
radii = distances.distance_array(
    coords, partial_shell.positions, box=u.dimensions
)[0]
```

Having that closing bracket on the same column as `radii` makes it really difficult for me to figure out where the definition for `radii` starts and ends. I would natively write something like

```
radii, *_ = distances.distance_array(coords,
									 partial_shell.positions,
									 box=u.dimensions)
```

There are other autoformatter tools but I mostly like how unobtrusive the pep8bot is. If I agree with its complaints, I change the formatting -- if not, I don't.

> which is PEP8 compliant

I also think you're actually using black in its native configuration which sets line lengths to 88. PEP8 is 79.

Reply:
Although I would also ask why you go to the trouble of computing the entire distance array if you only want the first column, wouldn't this be faster?

`radii = distances.distance_array(coords[0], partial_shell.positions, box=u.dimensions)`

Reply:
Re: autoformatting - I agree with @lilyminium 

My main take (kinda echo's @lilyminium's) is that at the end of the day you are responsible for ensuring that your code is readable and maintainable by folks that may not be you. Using any kind of automated tools is great if it saves you time, but it's your responsibility to check the outputs and make sure that you agree with them.

Since this is outside of core, I won't bring it up again unless it's really unreadable. I'll trust that any further formatting decisions conforms to your own "coding style" (as long as it conforms to whatever formatting standard you set).

With regards to the MDAnalysis core library we're a bit more stringent and generally will tell folks off for using autoformatters, but that's mainly because it tends to overwrite a ton of lines that aren't related to a given PR (re-writing history as it goes along).

Reply:
Thank you for the formatting feedback. Given the format of this project, I think readability is more important than saving formatting time. I recently started using Black and it definitely deviates from what I am used to. I think it's best to save it for projects that are built entirely using Black. For this, I'll give pep8bot a try or just stick to PyCharm's built in syntax suggestions. I'll go through and fix some of Black's more egregious formattting decisions in this file 🙂